src/main/java/org/folio/fqm/service/UserTenantService.java

```diff
@@ -0,0 +1,26 @@
+package org.folio.fqm.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.fqm.client.SimpleHttpClient;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+
+/**
+ * Service wrapper for caching responses from user-tenants API.
+ */
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class UserTenantService {
+
+  private final SimpleHttpClient userTenantsClient;
+
+  @Cacheable(value="userTenantCache", key="#tenantId")
+  public String getUserTenantsResponse(String tenantId) {
+    log.info("Retrieving user-tenants information for tenant {}", tenantId);
+    return userTenantsClient.get("user-tenants", Map.of("limit", String.valueOf(1)));
+  }
+}
```

Comment:
Added this class to allow for convenient caching for the user-tenants responses. We use the /user-tenants endpoint in both the EntityTypeFlatteningService and the CrossTenantQueryService, so I figured adding a service wrapper made the most sense to allow us to:
  1. Cache the user-tenants responses in a convenient way regardless of what class the request is made from.
  2. Avoid caching responses on any of the other SimpleHttpClient requests.